### PR TITLE
Added some pretty debugging to the FEM client.

### PR DESCRIPTION
**Pull request by suitupalex**

### CODE REVIEW

File: help-fem.js [hunk 1]
```diff
@@ -50,6 +50,15 @@
...
                : (this.inBrowser ? 'http://127.0.0.1:8080' : process.env.FEM);
 
+
+    // Toggle debugging
+    this.debug = options.debug ||
+      this.uri.indexOf('localhost') > -1 ||
+      this.uri.indexOf('127.0.0.1') > -1;
+
+    // Toggle expanding the debug messages
+    this.debugExpand = options.debugExpand || true;
```
Comment (nubs) [inline help-fem.js]: This will always be truthy

**Comment by suitupalex** [comment]
Reupped.

File: help-fem.js [hunk 2]
```diff
@@ -1202,6 +1232,22 @@
     // This method can be called to send any kind of message through Socket.io without
     // having to directly invoke the socket object.
     emit: function(type, payload) {
+      if (this.debug) {
```
Comment (nubs) [inline help-fem.js]: this duplicate code could easily be pulled out into a new method that takes three parameters: the category, the type, and the JSON.  That would make the emit and callback functions cleaner too.

**Comment by suitupalex** [comment]
Rereupped. Added a yellow color for misc. socket.io events.

**Comment by nubs** [comment]
nice :)  LGTM

**Pull request merged**